1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,208 @@ Bitcoin blockchain, such as exchanges or custodians:
+> valid permanently.
+
+Replacing a transaction does not invalidate the old transaction, so unless
+there is an incentive for the miner to use the new transaction, then just
+replacing an old transaction with one that has higher `nSequence` numbers
+is not incentive-compatible.
+
+## Replace-by-Fee
+
+Replace-by-fee is an effort to make transactions replaceable whilst avoiding
+the problems of exposing nodes to DoS vulnerabilities and misaligned
+incentives.  This is achieved by only allowing replacement if the replacement
+transaction has a higher fee than the replaced transaction. This prevents DoS
+attacks since every replacement transaction costs the attacker additional
+money, and ensures incentive alignment since the replacement transaction is
+financially advantageous to the miner.
+
+There are several [variants of replace-by-fee][rbf variants]. Almost all nodes
+on the Bitcoin are thought to implement 'opt-in RBF' as defined in [BIP 125][].
+
+Bitcoin Core may implement 'delayed RBF' at some point in the future. There
+is currently an [open PR][delayed rbf pr], which has support from several
+developers.
 
 ## BIP 125
 
-TODO
+BIP 125 is a specification for opt-in RBF written by David Harding and Peter
+Todd in December 2015. It describes the conditions under which an unconfirmed
+transaction in a node's mempool will be replaced. BIP 125 is referred to as
+*opt-in* RBF since the node will only allow a transaction to be replaced if
+the sender has explicitly signalled that he wants the transaction to be
+replacable.
+
+> WARNING: transaction replacement policies are outside the domain of consensus.
+> A transaction not being marked as replaceable offers no guarantees that it won't
+> be double-spent. Other nodes on the network may have different replacement
+> policies and miners are free to select whatever transactions they want in
+> blocks.
+
+The spender signals that a transaction is replaceable by setting the `nSequence`
+number on any of its inputs to less than (0xffffffff - 1). A transaction is
+replaceable if it fulfils [five conditions][]. The BIP documents those condtions,
+but the implications of those conditions may not be immediately obvious. The next
+section describes some of those implications.
+
+[five conditions]: https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#implementation-details
+
+### Implications of the conditions for BIP 125 RBF
+
+> One or more transactions currently in the mempool (original transactions)
+> will be replaced by a new transaction (replacement transaction) that spends
+> one or more of the same inputs if,
+
+- implication 1: RBF can be used to replace multiple transactions at the same
+  time. That could either be because the original transaction has descendants
+  in the mempool that would be invalidated by the replacement transaction and
+  so need to be removed, or because the replacement transaction is spending
+  inputs in more than one original transaction.
```

Comment:
@dgpv says "Maybe I'm missing something, but if we're replacing a tx that spends inputs from different parent txs, parent txs are not removed from mempool. Only the tx that we are replacing is removed.

So in this case, we would be replacing only one tx (if it does not have descendants itself)"

Reply:
Yes, it makes sense. Maybe it was harder for me to understand the meaning because the case about descendant transactions was presented first, and it created a sort of mental anchor.

I think it would be more clear to say it like this: ".. or because the new transaction just directly replaces two or more transactions by spending their inputs". Or even talk about this case first, and about the case with descendants later.

Reply:
That's fair. I've switched the order, so it now reads:

```
- implication 1: RBF can be used to replace multiple transactions at the same
  time. That could either be because the replacement transaction is directly
  replacing two or more original tranasactions by spending their inputs, or
  because the original transaction that is being replaced has descendants in
  the mempool which will also need to be removed.
```